apis/v1alpha1/secret.go

```diff
@@ -0,0 +1,199 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SecretSpec defines the desired state of Secret.
+type SecretSpec struct {
+
+	// A list of Regions and KMS keys to replicate secrets.
+	AddReplicaRegions []*ReplicaRegionType `json:"addReplicaRegions,omitempty"`
+	// If you include SecretString or SecretBinary, then Secrets Manager creates
+	// an initial version for the secret, and this parameter specifies the unique
+	// identifier for the new version.
+	//
+	// If you use the Amazon Web Services CLI or one of the Amazon Web Services
+	// SDKs to call this operation, then you can leave this parameter empty. The
+	// CLI or SDK generates a random UUID for you and includes it as the value for
+	// this parameter in the request. If you don't use the SDK and instead generate
+	// a raw HTTP request to the Secrets Manager service endpoint, then you must
+	// generate a ClientRequestToken yourself for the new version and include the
+	// value in the request.
+	//
+	// This value helps ensure idempotency. Secrets Manager uses this value to prevent
+	// the accidental creation of duplicate versions if there are failures and retries
+	// during a rotation. We recommend that you generate a UUID-type (https://wikipedia.org/wiki/Universally_unique_identifier)
+	// value to ensure uniqueness of your versions within the specified secret.
+	//
+	//   - If the ClientRequestToken value isn't already associated with a version
+	//     of the secret then a new version of the secret is created.
+	//
+	//   - If a version with this value already exists and the version SecretString
+	//     and SecretBinary values are the same as those in the request, then the
+	//     request is ignored.
+	//
+	//   - If a version with this value already exists and that version's SecretString
+	//     and SecretBinary values are different from those in the request, then
+	//     the request fails because you cannot modify an existing version. Instead,
+	//     use PutSecretValue to create a new version.
+	//
+	// This value becomes the VersionId of the new version.
+	ClientRequestToken *string `json:"clientRequestToken,omitempty"`
```

Comment:
Please add this to the ignored shapes in the `generator.yaml`

Reply:
ignore in [remove ClientRequestToken](https://github.com/aws-controllers-k8s/secretsmanager-controller/pull/2/commits/2c690bb18019fe96f9c7a3fad8ac9fcf10f85d41)